docs/en/latest/plugins/ai-proxy.md

```diff
@@ -0,0 +1,144 @@
+| model.options             | No           | Object   | Key/value settings for the model                                                     |
+| model.options.max_tokens  | No           | Integer  | Defines the max tokens if using chat or completion models. Default: 256              |
+| model.options.input_cost  | No           | Number   | Cost per 1M tokens in your prompt. Minimum: 0                                        |
+| model.options.output_cost | No           | Number   | Cost per 1M tokens in the output of the AI. Minimum: 0                               |
+| model.options.temperature | No           | Number   | Matching temperature for models. Range: 0.0 - 5.0                                    |
+| model.options.top_p       | No           | Number   | Top-p probability mass. Range: 0 - 1                                                 |
+| model.options.stream      | No           | Boolean  | Stream response by SSE. Default: false                                               |
+| model.override.endpoint   | No           | String   | Override the endpoint of the AI provider                                             |
+| passthrough               | No           | Boolean  | If enabled, the response from LLM will be sent to the upstream. Default: false       |
+| timeout                   | No           | Integer  | Timeout in milliseconds for requests to LLM. Range: 1 - 60000. Default: 3000         |
+| keepalive                 | No           | Boolean  | Enable keepalive for requests to LLM. Default: true                                  |
+| keepalive_timeout         | No           | Integer  | Keepalive timeout in milliseconds for requests to LLM. Minimum: 1000. Default: 60000 |
+| keepalive_pool            | No           | Integer  | Keepalive pool size for requests to LLM. Minimum: 1. Default: 30                     |
+| ssl_verify                | No           | Boolean  | SSL verification for requests to LLM. Default: true                                  |
+
+## Example usage
+
+Create a route with the `ai-proxy` plugin like so:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes/1" -X PUT \
+  -H "X-API-KEY: ${ADMIN_API_KEY}" \
+  -d '{
+    "uri": "/anything",
+    "plugins": {
+      "ai-proxy": {
+        "auth": {
+          "header": {
+            "Authorization": "Bearer <some-token>"
+          }
+        },
+        "model": {
+          "provider": "openai",
+          "name": "gpt-4",
+          "options": {
+            "max_tokens": 512,
+            "temperature": 1.0
+          }
+        }
+      }
+    },
+    "upstream": {
+      "type": "roundrobin",
+      "nodes": {
+        "somerandom.com:443": 1
+      },
+      "scheme": "https",
+      "pass_host": "node"
+    }
+  }'
+```
+
+Since `passthrough` is not enabled upstream node can be any arbitrary value because it won't be contacted.
```

Comment:
1. design question: why do we need the parameter `passthrough`? is it meant to be a toggle?
2. is this sentence needed here?

Reply:
> design question: why do we need the parameter passthrough? is it meant to be a toggle?

yes, kinda. when it's enabled, the response from LLM will be set as the request body to the upstream.

> is this sentence needed here?

I think it's important to let users know, instead to let them be puzzled about it and then figure out later.